app/controllers/subscriptions_controller.rb

```diff
@@ -61,6 +61,34 @@ def destroy
     end
   end
 
+  def confirm_delete_all
+    @subscribable_type = params[:type] if params[:type] && Subscription::VALID_SUBSCRIBABLES.include?(params[:type].singularize.titleize)
+  end
+
+  def delete_all
+    if params[:type] && Subscription::VALID_SUBSCRIBABLES.include?(params[:type].singularize.titleize)
+      @subscribable_type = params[:type]
+      @subscriptions = @user.subscriptions.where(subscribable_type: @subscribable_type.classify)
+    else
+      @subscriptions = @user.subscriptions
+    end
+
+    success = true
+    @subscriptions.each do |subscription|
+      subscription.destroy
```

Comment:
```suggestion
      subscription.destroy!
```
The other one won't throw if there is an issue, it'll just return false.